packages/organization/src/organization.ts

```diff
@@ -0,0 +1,144 @@
+import { type Organization, Prisma, Tier, prisma } from '@repo/database';
+import { maxUsersPerTier } from '@repo/mappings';
+import { revokeIntegration } from '@repo/channel-utils';
+import { FireAndForget } from '@repo/utils';
+
+const fireAndForget = new FireAndForget();
+
+export const getRootOrganizationId = async (organizationId: string): Promise<string> => {
+  const organization = await prisma.organization.findUniqueOrThrow({ where: { id: organizationId } });
+  if (organization.parentId) {
+    return await getRootOrganizationId(organization.parentId);
+  }
+  return organization.id;
+};
+
+export const getTier = async (organizationId?: string | null): Promise<Tier> => {
+  if (!organizationId) {
+    return Tier.Launch;
+  }
+
+  const rootOrganizationId = await getRootOrganizationId(organizationId);
+  const organization = await prisma.organization.findUniqueOrThrow({ where: { id: rootOrganizationId } });
```

Comment:
```suggestion
    const organization = await prisma.organization.findUniqueOrThrow({
    select: { tier: true }, where: { id: rootOrganizationId },
  });
```
This is jfyi to learn if you haven't already that you can do this on prisma. You may keep the code as is.